pypeit/pypeit.py

```diff
@@ -583,67 +593,35 @@ def extract_one(self, frames, det, bg_frames=[], std_outfile=None):
         # Prep for manual extraction (if requested)
         manual_extract_dict = self.fitstbl.get_manual_extract(frames, det)
 
-        # Do one iteration of object finding, and sky subtract to get initial sky model
-        self.sobjs_obj, self.nobj, skymask_init = \
-            self.redux.find_objects(self.sciImg.image, std=self.std_redux, ir_redux=self.ir_redux,
-                                    std_trace=std_trace, maskslits=self.maskslits,
-                                    show=self.show & (not self.std_redux),
-                                    manual_extract_dict=manual_extract_dict)
-
-        # Global sky subtraction, first pass. Uses skymask from object finding step above
-        self.initial_sky = \
-            self.redux.global_skysub(skymask=skymask_init,
-                                    std=self.std_redux, maskslits=self.maskslits, show=self.show)
-        if not self.std_redux:
-            # Object finding, second pass on frame *with* sky subtraction. Show here if requested
-            self.sobjs_obj, self.nobj, self.skymask = \
-                self.redux.find_objects(self.sciImg.image - self.initial_sky, std=self.std_redux,
-                                        ir_redux=self.ir_redux, std_trace=std_trace,maskslits=self.maskslits,
-                                        show=self.show, manual_extract_dict=manual_extract_dict)
-
-        # If there are objects, do 2nd round of global_skysub, local_skysub_extract, flexure, geo_motion
-        if self.nobj > 0:
-            # Global sky subtraction second pass. Uses skymask from object finding
-            self.global_sky = self.initial_sky if self.std_redux else \
-                self.redux.global_skysub(skymask=self.skymask, maskslits=self.maskslits, show=self.show)
-
-            self.skymodel, self.objmodel, self.ivarmodel, self.outmask, self.sobjs = \
-            self.redux.local_skysub_extract(self.caliBrate.mswave, self.global_sky, self.sobjs_obj,
-                                            model_noise=(not self.ir_redux),std = self.std_redux,
-                                            maskslits=self.maskslits, show_profile=self.show,show=self.show)
+        # Find objects
```

Comment:
I don't agree with this to reduce, at least in the way that you have done it, i.e. suppressing argument lists, and making it very difficult to follow the control flow. My proposal is that we restore the original control flow here, and then discuss how to move this code to reduce if you want it in reduce. Your new find_objects code is obfuscating what the code actually does and this already created a significant problem in coadd2d where you implemented object finding incorrectly.

Reply:
resolved.